gobblin-metrics/src/main/java/gobblin/metrics/reporter/EventReporter.java

```diff
@@ -62,7 +65,10 @@ public EventReporter(Builder builder) {
     super(builder.context, builder.name, builder.filter, builder.rateUnit, builder.durationUnit);
 
     this.closer = Closer.create();
-    this.immediateReportExecutor = Executors.newSingleThreadExecutor();
+    this.immediateReportExecutor = MoreExecutors.
+        getExitingExecutorService((ThreadPoolExecutor) Executors.newFixedThreadPool(1,
+            new ThreadFactoryBuilder().setNameFormat("EventReporter-" + builder.name + "-%d").build()),
+            5, TimeUnit.MINUTES);
```

Comment:
They should ideally shutdown immediately. 5 min is an arbitrary buffer in case some messages still need to be sent to Kafka.